### ISSUE TITLE
[template/i18n] reports error for ng-template's marked with i18n tag

### ISSUE THREAD

**Issue by mattlewis92** [angular-eslint/angular-eslint#701]
**Description and reproduction of the issue**

```JSON
{
  "rules": {
        "@angular-eslint/template/i18n": [
          "error",
          {
            "checkId": true,
            "checkAttributes": false,
            "ignoreTags": [":svg:title", ":svg:desc", ":svg:tspan"]
          }
        ],
  }
}
```

i18n tags on `ng-template` tags get marked as missing i18n e.g. both of these cases are reported as errors (via the latest canary version of angular-eslint):
```HTML
      <ng-template
        [ngIf]="userGroups?.length > 0"
        [ngIfElse]="noTeamsTmpl"
        i18n="@@cu-user-list__no-matching-users-groups"
      >
        No people or teams matched your search
      </ng-template>
      <ng-template i18n="@@cu-user-list__no-matching-users" #noTeamsTmpl>
        No people matched your search
      </ng-template>
```

This error also gets thrown for some libraries (disabling the i18n rule fixes it so it originates somewhere in that rule):

```
Cannot read property 'start' of undefined
```

I presume it's caused by one of these 2 arguments being `undefined`:
https://github.com/angular-eslint/angular-eslint/blob/master/packages/eslint-plugin-template/src/rules/i18n.ts#L337
https://github.com/angular-eslint/angular-eslint/blob/master/packages/eslint-plugin-template/src/rules/i18n.ts#L348

The stack trace is kind of useless, so I'm struggling to find the exact bit of code causing that error:

```
TypeError: Cannot read property 'start' of undefined
    at Linter._verifyWithoutProcessors (/Users/mattlewis/Code/clickup/frontend/node_modules/eslint/lib/linter/linter.js:1194:54)
    at Linter._verifyWithConfigArray (/Users/mattlewis/Code/clickup/frontend/node_modules/eslint/lib/linter/linter.js:1276:21)
    at Linter.verify (/Users/mattlewis/Code/clickup/frontend/node_modules/eslint/lib/linter/linter.js:1231:25)
    at Linter.verifyAndFix (/Users/mattlewis/Code/clickup/frontend/node_modules/eslint/lib/linter/linter.js:1424:29)
    at verifyText (/Users/mattlewis/Code/clickup/frontend/node_modules/eslint/lib/cli-engine/cli-engine.js:234:48)
    at CLIEngine.executeOnFiles (/Users/mattlewis/Code/clickup/frontend/node_modules/eslint/lib/cli-engine/cli-engine.js:802:28)
    at ESLint.lintFiles (/Users/mattlewis/Code/clickup/frontend/node_modules/eslint/lib/eslint/eslint.js:563:23)
    at Object.<anonymous> (/Users/mattlewis/Code/clickup/frontend/packages/linter/src/executors/eslint/utility/eslint-utils.ts:41:23)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/mattlewis/Code/clickup/frontend/node_modules/tslib/tslib.js:114:62)
```

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `12.5.1-alpha.5` |
| `@angular-eslint/template-parser`         | `12.5.1-alpha.5` |
| `@typescript-eslint/parser`               | `4.28.5` |
| `ESLint`                                  | `7.29.0` |
| `node`                                    | `14.17.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 12.2.6
Node: 14.17.0
Package Manager: npm 6.14.13
OS: darwin x64

Angular: 12.2.6
... animations, cli, common, compiler, compiler-cli, core
... elements, forms, language-service, localize
... platform-browser, platform-browser-dynamic, router
... service-worker

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1202.1
@angular-devkit/build-angular   12.2.6
@angular-devkit/core            12.1.4
@angular-devkit/schematics      12.2.6
@angular/cdk                    12.2.1
@angular/google-maps            12.2.1
@schematics/angular             12.2.6
ng-packagr                      12.2.1
rxjs                            7.3.0
typescript                      4.3.5
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.
